tests/test_vocab.py

```diff
@@ -0,0 +1,72 @@
+import os
+import tempfile
+
+import pytest
+import finalfusion.vocab
+
+from finalfusion.io import FinalfusionFormatError
+
+
+def test_reading(tests_root):
+    with pytest.raises(TypeError):
+        finalfusion.vocab.load_vocab(None)
+    with pytest.raises(FinalfusionFormatError):
+        # 0 opens sys.stdin, should result in an error when trying to read magic
+        finalfusion.vocab.load_vocab(0)
+    with pytest.raises(IOError):
+        finalfusion.vocab.load_vocab("foo")
+    vocab_path = os.path.join(tests_root, "data", "simple_vocab.fifu")
+    v = finalfusion.vocab.load_vocab(vocab_path)
+    assert v.words[0] == "Paris"
+
+
+def test_contains():
+    v = finalfusion.vocab.SimpleVocab([str(i) for i in range(10)])
+    assert "1" in v
+    assert None not in v
+    assert ["1", "2"] in v
+    assert {"1", "2"} in v
+    assert map(str, range(2)) in v
+    assert not range(2) in v
+
+
+def test_simple_roundtrip(tests_root, tmp_path):
+    filename = os.path.join(tmp_path, "write_simple.fifu")
```

Comment:
Things like `tmp_path / "write_simple.fifu"` should work. The same for other pathlib paths. Some people like to overload their operators :).

https://docs.python.org/3.8/library/pathlib.html#operators